src/Domain/Creators/TimeDependence/SphericalCompression.cpp

```diff
@@ -0,0 +1,128 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/Creators/TimeDependence/SphericalCompression.hpp"
+
+#include <array>
+#include <cstddef>
+#include <limits>
+#include <memory>
+#include <optional>
+#include <ostream>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/IndexType.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.tpp"
+#include "Domain/CoordinateMaps/MapInstantiationMacros.hpp"
+#include "Domain/Creators/TimeDependence/TimeDependence.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/FunctionsOfTime/PiecewisePolynomial.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace domain {
+namespace creators::time_dependence {
+
+SphericalCompression::SphericalCompression(
+    const double initial_time,
+    const std::optional<double> initial_expiration_delta_t,
+    const double min_radius, const double max_radius,
+    const std::array<double, 3> center, const double initial_value,
+    const double initial_velocity, const double initial_acceleration,
+    std::string function_of_time_name) noexcept
+    : initial_time_(initial_time),
+      initial_expiration_delta_t_(initial_expiration_delta_t),
+      min_radius_(min_radius),
+      max_radius_(max_radius),
+      center_(center),
+      initial_value_(initial_value),
+      initial_velocity_(initial_velocity),
+      initial_acceleration_(initial_acceleration),
+      function_of_time_name_(std::move(function_of_time_name)) {
+  if (min_radius >= max_radius) {
+    ERROR(
```

Comment:
you can just add the `OptionsContext` as a constructor argument. In the header `const OptionContext& context = {}` and then just `const OptionContext& context` in the cpp